### PR TITLE
eeprom: Add EEPROM over SMBus simple driver

### PR DESCRIPTION
**Pull request by finikorg**
Add simple EEPROM over SMBus driver, support for Qemu Q35 platform EEPROMs and include to EEPROM testing.

I included Qemu EEPROMs to general DTS, can move them to sample / test overlays.

### CODE REVIEW

**Review by jhedberg — APPROVED** [review]

File: dts/bindings/mtd/zephyr,smbus-eeprom.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2022,2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+description: Zephyr SMBus EEPROM binding
+
+compatible: "zephyr,smbus-eeprom"
```
Comment (henrikbrixandersen) [inline dts/bindings/mtd/zephyr,smbus-eeprom.yaml:6]: Why the `zephyr,` prefix? Don't these devices represent actual hardware?

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
AFAICT, SMBus EEPROMs should be compatible with the existing AT24 EEPROM driver, except for the fact that we choose to implement SMBus support as a separate API instead of an API associated with the I2C API. The AT24 driver already support both AT24 (I2C) and  AT25 (SPI) EEPROMs. Would it make sense to extend it with SMBus EEPROM support as well?

I am curious - will the decision of having a completely separate API for SMBus lead to having duplicated driver efforts for basically the same devices as we support on the I2C bus?

File: dts/bindings/mtd/zephyr,smbus-eeprom.yaml (hunk 1 above)
Comment (finikorg) [reply dts/bindings/mtd/zephyr,smbus-eeprom.yaml:6]: This is driver for Qemu EEPROM "kind of" devices, I used it for testing SMBus only.
Source:
https://github.com/zephyrproject-rtos/qemu/blob/2cc2e86de6e0e4316265d34e8e935afade8c422e/hw/i386/pc_q35.c#L315
https://github.com/zephyrproject-rtos/qemu/blob/zephyr-qemu-v7.0.0/hw/i2c/smbus_eeprom.c

Comment (finikorg) [reply dts/bindings/mtd/zephyr,smbus-eeprom.yaml:6]: @henrikbrixandersen using heavy macrobatic I can make the AT24 EEPROM driver even more unreadable including SMBus part, the current code heavily uses `#idef`s to separate SPI and I2C parts.

**Comment by carlescufi** [comment]
Architecture WG:

- @henrikbrixandersen mentions that we already have an in-tree driver for EEPROM. This driver is a duplicate of the existing driver, just over SMBus instead of I2C. This driver uses the SMBus API instead of the I2C API
- @finikorg @jhedberg let's defer to the dev-review meeting this week

**Comment by finikorg** [comment]
@henrikbrixandersen I have added draft PR according to your suggestions: https://github.com/zephyrproject-rtos/zephyr/pull/57673

**Comment by finikorg** [comment]
Closing PR as the driver was used only for testing, no real device.

**Pull request closed without merging**